File: Examen_Corto/Octave/ejercicio2.m





num_str = input("Ingrese un número: ", "s");
num = str2num(num_str);

divisores = [];
for i = 1:num
    if mod(num, i) == 0
        divisores = [divisores, i];
    end
end

% Mostrar los divisores
divisores_str = sprintf("Los divisores de %d son: %s", num, num2str(divisores));
disp(divisores_str);


fid = fopen("ejercicio2.txt", "w");
fprintf(fid, "%s", divisores_str);
fclose(fid);



% Conexión a la base de datos PostgreSQL
pkg load database %cargar el paquete

conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
'port','5432','user','postgres','password','gabrielgrdb'))

salida = divisores_str;

query = sprintf("insert into salida values ('%s');", salida);
N = pq_exec_params(conn, query);

File: Examen_Corto/Python/Ejercicio2.py
import psycopg2

num_str = input("Ingrese un número: ")
num = int(num_str)

divisores = [i for i in range(1, num + 1) if num % i == 0]

# Mostrar los divisores
divisores_str = "Los divisores de {} son: {}".format(num, ', '.join(map(str, divisores)))
print(divisores_str)

# Escribir en el archivo
with open("ejercicio2.txt", "w") as file:
    file.write(divisores_str)

# Conexión a la base de datos PostgreSQL
try:
    conn = psycopg2.connect(
        dbname='ExamenCorto1',
        user='postgres',
        password='gabrielgrdb',
        host='localhost',
        port='5432'
    )

    cursor = conn.cursor()

    salida = divisores_str

    query = "insert into salida values (%s);"
    cursor.execute(query, (salida,))

    conn.commit()

except Exception as e:
    print("Error de conexión:", e)

finally:
    if conn is not None:
        conn.close()
